006_buildkite_importing_pipelines/README.md

```diff
@@ -0,0 +1,374 @@
+# Driving Buildkite Pipelines with CUE
+<sup>by [Jonathan Matthews](https://jonathanmatthews.com)</sup>
+
+This guide explains how to convert Buildkite pipeline files from YAML to CUE,
+check those pipelines are valid, and then use CUE's tooling layer to regenerate
+YAML.
+
+This allows you to switch to CUE as a source of truth for Buildkite pipelines
+and perform client-side validation, without Buildkite needing to know you're
+managing your pipelines with CUE.
+
+## Prerequisites
+
+- You have
+  [CUE installed](https://alpha.cuelang.org/docs/introduction/installation/)
+  locally. This allows you to run `cue` commands.
+- You have a set of Buildkite pipeline files. The examples shown in this guide
+  use a specific commit from the Buildkite
+  [dependent-pipeline-example](https://github.com/buildkite/dependent-pipeline-example/tree/a7419a24bee24068d4d79399bccd9093569c3013/.buildkite)
+  repository, *but you don't need to use that repository in any way.*
+- You have [`git` installed](https://git-scm.com/downloads).
+<!-- TODO: curl isn't needed until the doc includes fetching the upstream schema
+- You have [`curl` installed](https://curl.se/dlwiz/), or can fetch a file from
+  a website some other way.
+-->
+
+## Steps
+
+### Convert YAML pipelines to CUE
+
+#### :arrow_right: Begin with a clean git state
+
+Change directory into the root of the repository that contains your Buildkite
+pipeline files, and ensure you start this process with a clean git state, with
+no modified files. For example:
+
+:computer: `terminal`
+```sh
+cd dependent-pipeline-example # our example repository
+git status                    # should report "working tree clean"
+```
+
+#### :arrow_right: Initialise a CUE module
+
+Initialise a CUE module named after the organisation and repository you're
+working with. For example:
+
+:computer: `terminal`
+```sh
+cue mod init github.com/buildkite/dependent-pipeline-example
+```
+
+#### :arrow_right: Import YAML pipelines
+
+Use `cue` to import your YAML pipeline files:
+
+:computer: `terminal`
+```sh
+cue import ./.buildkite/*.yml --with-context -p buildkite -f -l pipelines: -l 'strings.TrimSuffix(path.Base(filename),path.Ext(filename))'
+```
+
+Check that a CUE file has been created for each YAML pipeline in the
+`.buildkite` directory. For example:
+
+:computer: `terminal`
+```sh
+ls .buildkite/
+```
+
+Your output should look similar to this, with matching pairs of YAML and CUE
+files:
+
+```text
+pipeline.cue pipeline.deploy.cue pipeline.deploy.yml pipeline.yml
```

Comment:
Same comment from other PR about file being name with `.` - looks like they are special suffixes as opposed to filenames.

Reply:
Done. I've called out the fact that the YAML files are pre-existing, and clarified that the pipeline file's name defines the pipeline's data's location in the `pipelines` struct.